integration/util/requirement/requirement.go

```diff
@@ -0,0 +1,29 @@
+package requirement
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// HasHubConnectivity checks to see if https://hub.docker.com is
+// accessible from the present environment
+func HasHubConnectivity() bool {
+	// Set a timeout on the GET at 15s
+	var timeout = 15 * time.Second
+	var url = "https://hub.docker.com"
+
+	client := http.Client{
+		Timeout: timeout,
+	}
+
+	resp, err := client.Get(url)
+	if err != nil && strings.Contains(err.Error(), "use of closed network connection") {
+		panic(fmt.Sprintf("Timeout for GET request on %s", url))
```

Comment:
It seems strange to panic here, especially on a timeout.

How about add a `t LogT` argument to this function and do this instead:

```go
t.Logf("Timeout ...")
return false
```

Reply:
This comes directly from <https://github.com/moby/moby/blob/master/integration-cli/requirements_test.go#L106> which I did not write. I don't think tests that require hub connectivity should allow test runs to pass by skipping them and emitting a logging message that will be missed/ignored in the thousands of lines of spew.

Reply:
We should fix these problem instead of carrying them over into the new test suite.

> I don't think tests that require hub connectivity should allow test runs to pass by skipping them and emitting a logging message that will be missed/ignored in the thousands of lines of spew.

If you think this should be a fatal error you can use `t.Fatal()`, but the purpose of these helpers is to detect state and skip tests that can't be run, so I don't think that's really appropriate. Skipping the test is the right behaviour. There is no reason to panic here.